onadata/apps/logger/signals.py

```diff
@@ -1,9 +1,44 @@
 # coding: utf-8
 import logging
-from django.db.models.signals import pre_delete
+
+from django.db.models import F
+from django.db.models.signals import (
+    pre_delete,
+    post_save,
+)
 from django.dispatch import receiver
 
 from onadata.apps.logger.models.attachment import Attachment
+from onadata.apps.logger.models.xform import XForm
+from onadata.apps.main.models.user_profile import UserProfile
+
+
+@receiver(pre_delete, sender=Attachment)
+def delete_user_profile_attachment_storage_bytes(instance, **kwargs):
+    """
+    Update the attachment_storage_bytes field in the UserProfile model
+    when an attachment is deleted
+    """
```

Comment:
[there's not a great way](https://github.com/kobotoolbox/kobocat/commit/9979941838c65dbec3da76db641c8c85de86c6ba) to deal with these signal handlers requiring their first argument to be called `instance` :cry: but maybe to help avoid confusion, you could add after the docstring something like `attachment = instance`. then, later, you could say `attachment.instance` instead of `instance.instance`